### PR TITLE
Istio 1.7: documentation update

### PR DESCRIPTION
**Pull request by colunira**
**Description**

Changes proposed in this pull request:

- Update info about istio 

**Related issue(s)**
#9503

### CODE REVIEW

File: docs/service-mesh/03-04-sidecar-proxy-injection.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ title: Sidecar Proxy Injection
 type: Details
 ---
 
-By default, the Istio sidecar injector watches all Pod creation operations on all Namespaces and injects the newly created Pods with a sidecar proxy.
+By default, the Istio watches all Pod creation operations on all Namespaces and injects the newly created Pods with a sidecar proxy.
```
Comment (Demonsthere) [inline docs/service-mesh/03-04-sidecar-proxy-injection.md:6]:
```suggestion
By default, the Istiod watches all Pod creation operations on all Namespaces and injects the newly created Pods with a sidecar proxy.
```

**Review by kubadz — COMMENTED** [review]
Please remove also the `CAUTION` in 03-03-istio-rbac.md file or the whole file, as it does not serve any important role currently. 
If there would be a global Istio RBAC configuration in Kyma it would be mentioned in `## Kyma-specific configuration` section of 03-01-istio-setup-in-kyma.md file.

File: docs/service-mesh/03-03-sidecar-proxy-injection.md [hunk 2]
```diff
@@ -3,7 +3,7 @@ title: Sidecar Proxy Injection
 type: Details
 ---
 
-By default, the Istio sidecar injector watches all Pod creation operations on all Namespaces and injects the newly created Pods with a sidecar proxy.
+By default, the Istiod watches all Pod creation operations on all Namespaces and injects the newly created Pods with a sidecar proxy.
```
Comment (majakurcius) [inline docs/service-mesh/03-03-sidecar-proxy-injection.md:6]:
```suggestion
By default, `istiod` watches all Pod creation operations on all Namespaces and injects the newly created Pods with a sidecar proxy.
```
I gather you mean the [Istio daemon](https://istio.io/latest/blog/2020/istiod/) here and it's not a typo, right? Just making sure.

**Review by majakurcius — DISMISSED** [review]

**Review by piotrmsc — APPROVED** [review]

File: docs/migration-guides/1.16-1.17.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+## Istio
+
+Kyma 1.17 comes with a new Istio version: 1.7.3. 
+
+Make sure you cluster runs Kubernetes version 1.16 or higher.
+
+Authenication Policy is no longer supported. Replace it with Peer Authentication and RequestAuthentication.
+
+Additionally, this Istio version updates the CRD of the Virtual Service. The way of providing CORS configuration changes:
```
Comment (majakurcius) [inline docs/migration-guides/1.16-1.17.md:11]:
```suggestion
Additionally, this Istio version updates the CRD of the Virtual Service. The way of providing cross-origin resource sharing (CORS) configuration changes:
```

File: docs/migration-guides/1.16-1.17.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+# Old way
+allowOrigin:
+  - "*"
+
+# New way
+allowOrigins:
+  - regex: ".*"
+```
+
+To resolve the compability issues, we introduce a migration job, which adds a new field to every Virtual Service on the cluster, without deleting the previous configuration:
```
Comment (majakurcius) [inline docs/migration-guides/1.16-1.17.md:23]:
```suggestion
To resolve the compability issues, we introduce a migration job which adds a new field to every Virtual Service on the cluster without deleting the previous configuration:
```

File: docs/migration-guides/1.16-1.17.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+```bash
+spec:
+  http:
+  - match:
+    corsPolicy:
+      allowOrigins:
+        - regex: ".*"
+```
+
+After the upgrade, check if you have a custom CORS allowed origins configuration in your Virtual Services. You will have to migrate them manually. Run:
```
Comment (majakurcius) [inline docs/migration-guides/1.16-1.17.md:34]:
```suggestion
After the upgrade, check if you have a custom Allowed Origins CORS configuration in your Virtual Services. You will have to migrate them manually. Run:
```

File: docs/migration-guides/1.16-1.17.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+```bash
+kubectl get vs --all-namespaces -o=go-template --template='{{ range .items }}{{ $name := .metadata.name}}{{ range $key, $value := .spec.http}}{{if $value.corsPolicy.allowOrigin }}{{ printf "%s : " $name }}{{ printf "%v\n" $value.corsPolicy.allowOrigin }}{{println ""}}{{ end }}{{ end }}{{ end }}'
+```
+
+The command returns a list of pairs consisting of Virtual Service name and allowed origins. Look for entries different than `{VS_NAME} : [*]` and migrate them manually to the [new version](https://istio.io/latest/docs/reference/config/networking/virtual-service/#CorsPolicy). You can find the old configuration under the key `spec.http.corsPolicy.allowOrigin`.
```
Comment (majakurcius) [inline docs/migration-guides/1.16-1.17.md:40]:
```suggestion
The command returns a list of pairs consisting of the Virtual Service name and allowed origins. Look for entries different from `{VS_NAME} : [*]` and migrate them manually to the [new version](https://istio.io/latest/docs/reference/config/networking/virtual-service/#CorsPolicy). You can find the old configuration under the key `spec.http.corsPolicy.allowOrigin`.
```

File: docs/migration-guides/1.16-1.17.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+```bash
+spec:
+  http:
+  - match:
+    corsPolicy:
+      allowOrigins:
+        - regex: ".*"
+```
+
+After the upgrade, check if you have a custom Allowed Origins CORS configuration in your Virtual Services. You will have to migrate them manually. Run:
```
Comment (majakurcius) [inline docs/migration-guides/1.16-1.17.md:34]:
```suggestion
After the upgrade, check if you have a custom Allowed Origins CORS configuration in your Virtual Services. You will have to migrate it manually. Run:
```

**Review by majakurcius — APPROVED** [review]

File: docs/migration-guides/1.16-1.17.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
+# Migrate from 1.15 to 1.16
+
+## Istio
+
+Kyma 1.17 comes with a new Istio version: 1.7.3.
```
Comment (kubadz) [inline docs/migration-guides/1.16-1.17.md:5]:
```suggestion
Kyma 1.17 comes with a new Istio version: 1.7.4. 
```

**Review by majakurcius — APPROVED** [review]

File: docs/migration-guides/1.15-1.17.md [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+Also, the `container` label for a function pointing to `lambda` in the previous releases now points to `function`. 
+If you search for the logs from a function with the `container: lambda` filter, you will need to change it to `container: function` from this release on.
+
+## Migrate from 1.16.0-rc3 to 1.17
+
+### Istio
+
+Kyma 1.17 comes with a new Istio version: 1.7.4. 
+
+Make sure you cluster runs Kubernetes version 1.16 or higher.
```
Comment (alexandra-simeonova) [inline docs/migration-guides/1.15-1.17.md:66]:
```suggestion
>**NOTE:** To upgrade successfully, make sure your cluster runs Kubernetes version 1.16 or higher.
```

File: docs/migration-guides/1.15-1.17.md [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+
+Authenication Policy is no longer supported. Replace it with Peer Authentication and RequestAuthentication.
```
Comment (alexandra-simeonova) [inline docs/migration-guides/1.15-1.17.md:68]:
```suggestion
Istio's Authenication Policy is no longer supported. Replace it with Peer Authentication and RequestAuthentication.
```

File: docs/migration-guides/1.15-1.17.md [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+
+Additionally, this Istio version updates the CRD of the Virtual Service. The way of providing cross-origin resource sharing (CORS) configuration changes:
```
Comment (alexandra-simeonova) [inline docs/migration-guides/1.15-1.17.md:70]:
```suggestion
Additionally, this Istio version updates the CRD of the Virtual Service. The way to provide cross-origin resource sharing (CORS) configuration changes:
```

File: docs/service-mesh/03-03-sidecar-proxy-injection.md [hunk 12]
```diff
@@ -3,7 +3,7 @@ title: Sidecar Proxy Injection
 type: Details
 ---
 
-By default, the Istio sidecar injector watches all Pod creation operations on all Namespaces and injects the newly created Pods with a sidecar proxy.
+By default, istiod watches all Pod creation operations on all Namespaces and injects the newly created Pods with a sidecar proxy.
```
Comment (alexandra-simeonova) [inline docs/service-mesh/03-03-sidecar-proxy-injection.md:6]:
```suggestion
By default, `istiod` watches all Pod creation operations on all Namespaces and injects the newly created Pods with a sidecar proxy.
```

**Review by alexandra-simeonova — APPROVED** [review]

**Pull request merged**